### DISCUSSION TITLE
defineEventHandler should support multiple handlers like express

### DISCUSSION THREAD

**Discussion by notshekhar** [nitrojs/nitro#314] [Ideas]
Express: app.get(handler1, handler2, handler3)
Nitro should also be like: defineEventHandler(handler1, handler2, handler3)

**Comment by pi0** [comment]
Please check my answer in h3: https://github.com/unjs/h3/issues/127#issuecomment-1164989158

**Comment by notshekhar** [comment]
yeah its working thank you @pi0 
I have also created a custom hook which takes multiple handlers 
```
import { createApp } from "h3"

export default function defineEventHandlers() {
    const app = createApp()
    app.use([...arguments].map((func) => defineEventHandler(func)))
    return app
}
```

**Reply by pi0** [reply]
Perfect but you probably don't want to call `createApp` in each request. Your code example does that. Use `defineLazyEventHandler` if need an async logic to initialize dynamic middleware:

```js
export default function defineLazyEventHandler () {
    const app = createApp()
    app.use([...arguments].map((func) => defineEventHandler(func)))
    return app
}
```

**Reply by notshekhar** [reply]
@pi0 https://stackblitz.com/edit/github-rqbwda-kf4p8s?file=defineEventHandlers.ts,routes%2Findex.ts

**Reply by notshekhar** [reply]
can you please edit this code and show me how can I do this, will be very helpful
https://stackblitz.com/edit/github-rqbwda-kf4p8s?file=defineEventHandlers.ts,routes%2Findex.ts

**Reply by pi0** [reply]
Here you are made some improvements: https://stackblitz.com/edit/github-rqbwda-pycuae?file=defineEventHandlers.ts,routes%2Findex.ts

- Little bit of verbosity of calling `eventHandler` in Array, gives you full type support
- Using `lazyEventHandler`, lazily creates the app but also you can make it async to initialize handlers.

**Reply by notshekhar** [reply]
@pi0 Please add this hook to h3 too. It will be great if  h3 will support multiple handlers
